File: 周赛/【305周赛】6136-【优】算数三元组的数目.py
class Solution:
    def arithmeticTriplets(self, nums: List[int], diff: int) -> int:
        n = len(nums)
        ans = 0
        for i in range(n):
            if nums[i]-diff in nums and nums[i]+diff in nums:
                ans += 1
        return ans

File: 周赛/【305周赛】6136-【优】算数三元组的数目.cpp
class Solution {
public:
    int arithmeticTriplets(vector<int>& nums, int diff) {
        int n=nums.size();
        unordered_set<int> st;
        for(int x:nums)
            st.insert(x);
        int ans=0;
        for(int a:nums)
            if(st.count(a-diff)>0 && st.count(a+diff)>0)
                ans++;
        return ans;
    }
};
